hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/QueryParser.java

```diff
@@ -46,43 +45,32 @@
 
     private final HazelcastTypeFactory typeFactory;
     private final CatalogReader catalogReader;
-    private final SqlConformance conformance;
     private final SqlConformance jetConformance;
     private final List<Object> arguments;
 
-    private final SqlBackend sqlBackend;
     private final SqlBackend jetSqlBackend;
 
     public QueryParser(
             HazelcastTypeFactory typeFactory,
             CatalogReader catalogReader,
-            SqlConformance conformance,
             SqlConformance jetConformance,
             List<Object> arguments,
-            @Nonnull SqlBackend sqlBackend,
             @Nullable SqlBackend jetSqlBackend
     ) {
         this.typeFactory = typeFactory;
         this.catalogReader = catalogReader;
-        this.conformance = conformance;
         this.jetConformance = jetConformance;
         this.arguments = arguments;
-
-        this.sqlBackend = sqlBackend;
         this.jetSqlBackend = jetSqlBackend;
     }
 
     public QueryParseResult parse(String sql) {
         try {
-            try {
-                return parse(sql, sqlBackend, conformance);
-            } catch (Exception e) {
-                // TODO: once IMDG engine is removed, move the check (and fail fast) to SqlServiceImpl?
-                if (jetSqlBackend != null) {
-                    return parse(sql, jetSqlBackend, jetConformance);
-                } else {
-                    throw e;
-                }
+            if (jetSqlBackend != null) {
+                return parse(sql, jetSqlBackend, jetConformance);
+            } else {
+                // TODO: move this check to SqlServiceImpl.
+                throw new RuntimeException("Jet is disabled, see JetConfig#setEnabled.");
```

Comment:
```suggestion
                throw new JetDisabledException(JET_IS_DISABLED_MESSAGE);
```
I recommend doing this in order to make this consistent with #19259